Automattic/jetpack
# Search: migrate search widget to package

#### Changes proposed in this Pull Request:
The PR moves Jetpack Search widget to the search package. Changes include:

- Moved assets to the search package.
- Text domain changed to `jetpack-search-pkg`
- Moved initialization to `Jetpack_Initializer`
- Some other changes regarding namespace change

PHP linter tried to arrange the order of the functions which lead to possibly hard review, and had to re-order and commit with `--no-verify`.

#### Jetpack product discussion
pcNPJE-oI-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Navigate to a site with Jetpack Search subscription
- Disable Instant search
- Add Jetpack Search widget to sidebar
- Ensure assets load on the frontend: `search-widget-frontend.js` and `search-widget-frontend.css` (the style file is concatenated in /wp-content/plugins/jetpack/css/jetpack.css)
- Ensure Jetpack Search widget works in widget editor and customizer, meaning you could add/remove filters and save changes
- Ensure Jetpack Search widget works on the frontend for both Instant and Classic Search

Note: please use a theme that supports  widgets (e.g. theme twentytwenty) as FSE themes don't have widget areas.
